crates/ruff_linter/src/rules/flake8_pytest_style/rules/raises.rs

```diff
@@ -151,6 +151,53 @@ impl Violation for PytestRaisesWithoutException {
     }
 }
 
+/// ## What it does
+/// Checks for non-raw literal string arguments passed to the `match` parameter
+/// of `pytest.raises()` that contains at least one regex metacharacters.
+///
+/// ## Why is this bad?
+/// The `match` argument is implicitly converted to a regex under the hood.
+/// It should be made explicit whether the string is meant to be a regex or a "plain" pattern
+/// by either prefixing the string with the `r` suffix or wrapping it in `re.escape()`.
+///
+/// ## Example
+///
+/// ```python
+/// with pytest.raises(Exception, match="foo."):
+///     do_thing_that_raises()
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// with pytest.raises(Exception, match=r"foo."):
+///     do_thing_that_raises()
+/// ```
+///
+/// Alternatively:
+///
+/// ```python
+/// import re
+///
+///
+/// with pytest.raises(Exception, match=re.escape("foo.")):
+///     do_thing_that_raises()
+/// ```
```

Comment:
````suggestion
/// ```python
/// import pytest
///
///
/// with pytest.raises(Exception, match="A full sentence."):
///     do_thing_that_raises()
/// ```
///
/// Use instead:
///
/// ```python
/// import pytest
///
///
/// with pytest.raises(Exception, match=r"A full sentence."):
///     do_thing_that_raises()
/// ```
///
/// Alternatively:
///
/// ```python
/// import pytest
/// import re
///
///
/// with pytest.raises(Exception, match=re.escape("A full sentence.")):
///     do_thing_that_raises()
/// ```
///
/// or:
///
/// ```python
/// import pytest
/// import re
///
///
/// with pytest.raises(Exception, "A full sentence\\."):
///     do_thing_that_raises()
/// ```
````

Reply:
It would also be great to add a `## References` section at the bottom, with links to the docs for `pytest.raises()` and `re.escape()`